layers/ip6.go

```diff
@@ -34,6 +75,7 @@ type IPv6 struct {
 	HopLimit     uint8
 	SrcIP        net.IP
 	DstIP        net.IP
+	RoutingDstIP net.IP
```

Comment:
It'd have to return the final destination of the routing header, which is not a concern at all once the header is found in the payload.

Having one layer parse the others doesn't feel quite right yet but maybe it'll work. I'll give it some more thought.